larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.cc

```diff
@@ -0,0 +1,479 @@
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewThreeDKinkTool::PassesElementCuts(MatrixType::ElementList::const_iterator eIter, const ClusterSet &usedClusters) const
+{
+    if (usedClusters.count(eIter->GetCluster1()) || usedClusters.count(eIter->GetCluster2()))
+        return false;
+
+    if (eIter->GetOverlapResult().GetTwoViewXOverlap().GetXOverlapFraction0() - m_minXOverlapFraction < -std::numeric_limits<float>::epsilon())
+        return false;
+
+    if (eIter->GetOverlapResult().GetTwoViewXOverlap().GetXOverlapFraction1() - m_minXOverlapFraction < -std::numeric_limits<float>::epsilon())
+        return false;
+
+    if (eIter->GetOverlapResult().GetMatchingScore() - m_minMatchingScore < std::numeric_limits<float>::epsilon())
+        return false;
+
+    if (eIter->GetOverlapResult().GetLocallyMatchedFraction() - m_minLocallyMatchedFraction < std::numeric_limits<float>::epsilon())
+        return false;
+
+    return true;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+float TwoViewThreeDKinkTool::GetXSamplingPoint(const CartesianVector &splitPosition1, const bool isForwardInX,
+    const TwoDSlidingFitResult &fitResult1, const TwoDSlidingFitResult &fitResult2) const
+{
+    // Nearest common x position
+    float xMin1(std::numeric_limits<float>::max()), xMax1(-std::numeric_limits<float>::max());
+    float xMin2(std::numeric_limits<float>::max()), xMax2(-std::numeric_limits<float>::max());
+    fitResult1.GetMinAndMaxX(xMin1, xMax1);
+    fitResult2.GetMinAndMaxX(xMin2, xMax2);
+
+    const float commonX(isForwardInX ? std::max(xMin1, xMin2) : std::min(xMax1, xMax2));
+
+    if (isForwardInX && ((commonX > xMax1) || (commonX > xMax2)))
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    if (!isForwardInX && ((commonX < xMin1) || (commonX < xMin2)))
+        throw StatusCodeException(STATUS_CODE_NOT_FOUND);
+
+    // Layer step x position
+    float rL1(0.f), rT1(0.f);
+    fitResult1.GetLocalPosition(splitPosition1, rL1, rT1);
+    const int splitLayer(fitResult1.GetLayer(rL1));
+
+    const int lowLayer(std::max(fitResult1.GetMinLayer(), std::min(fitResult1.GetMaxLayer(), splitLayer - m_nLayersForKinkSearch)));
+    const int highLayer(std::max(fitResult1.GetMinLayer(), std::min(fitResult1.GetMaxLayer(), splitLayer + m_nLayersForKinkSearch)));
+
+    CartesianVector minus(0.f, 0.f, 0.f), plus(0.f, 0.f, 0.f);
+    PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, fitResult1.GetGlobalFitPosition(fitResult1.GetL(lowLayer), minus));
+    PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, fitResult1.GetGlobalFitPosition(fitResult1.GetL(highLayer), plus));
+
+    if (minus.GetX() > plus.GetX())
+    {
+        CartesianVector temporary(minus);
+        minus = plus;
+        plus = temporary;
```

Comment:
There's std::swap(minus, plus) function that will handle this functionality for you

Reply:
It's still [here](https://github.com/PandoraPFA/LArContent/blob/master/larpandoracontent/LArThreeDReco/LArTransverseTrackMatching/ThreeDKinkBaseTool.cc#L90-L95) too, but we do probably want to leave the three-view cases alone in this PR.